pkg/kubectl/cmd/get.go

```diff
@@ -182,13 +183,22 @@ func RunGet(f cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, args [
 		return cmdutil.UsageError(cmd, usageString)
 	}
 
-	// always show resources when getting by name or filename
+	// determine if args contains "all"
+	for _, a := range args {
+		if a == "all" {
```

Comment:
ug, it looks like that code has the same issue. I am not sure if there is a great solution:

https://github.com/kubernetes/kubernetes/blob/6e5b455b7762f73f599dbec69c9ceb7cb4c8dfca/pkg/kubectl/resource/builder.go#L808